pkg/cmd/info.go

```diff
@@ -25,41 +25,42 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// infoCmd represents the info command
-var infoCmd = &cobra.Command{
-	Use:   "info",
-	Short: "Get landscape informations\n",
-	Long:  ``,
-	Run: func(cmd *cobra.Command, args []string) {
-		var t Target
-		targetFile, err := ioutil.ReadFile(pathTarget)
-		checkError(err)
-		err = yaml.Unmarshal(targetFile, &t)
-		checkError(err)
-		if len(t.Target) < 1 {
-			fmt.Println("No garden targeted")
-			os.Exit(2)
-		}
-		// show landscape
-		Client, err = clientToTarget("garden")
-		gardenClientset, err := clientset.NewForConfig(NewConfigFromBytes(*kubeconfig))
-		checkError(err)
-		shootList, err := gardenClientset.GardenV1beta1().Shoots("").List(metav1.ListOptions{})
-		checkError(err)
-		fmt.Printf("Garden: %s\n", t.Target[0].Name)
-		fmt.Printf("Shoots:\n")
-		fmt.Printf("    total: %d \n", len(shootList.Items))
-		seedList, err := gardenClientset.GardenV1beta1().Seeds().List(metav1.ListOptions{})
-		for _, seed := range seedList.Items {
-			numberShootsPerInfrastructure := 0
-			for _, shoot := range shootList.Items {
-				if *shoot.Spec.Cloud.Seed == seed.Name {
-					numberShootsPerInfrastructure++
+// NewInfoCmd returns a new info command.
+func NewInfoCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "info",
+		Short: "Get landscape informations\n",
+		Run: func(cmd *cobra.Command, args []string) {
+			var t Target
+			targetFile, err := ioutil.ReadFile(pathTarget)
+			checkError(err)
+			err = yaml.Unmarshal(targetFile, &t)
+			checkError(err)
+			if len(t.Target) < 1 {
+				fmt.Println("No garden targeted")
+				os.Exit(2)
+			}
+			// show landscape
+			Client, err = clientToTarget("garden")
+			gardenClientset, err := clientset.NewForConfig(NewConfigFromBytes(*kubeconfig))
+			checkError(err)
+			shootList, err := gardenClientset.GardenV1beta1().Shoots("").List(metav1.ListOptions{})
+			checkError(err)
+			fmt.Printf("Garden: %s\n", t.Target[0].Name)
+			fmt.Printf("Shoots:\n")
+			fmt.Printf("    total: %d \n", len(shootList.Items))
+			seedList, err := gardenClientset.GardenV1beta1().Seeds().List(metav1.ListOptions{})
+			for _, seed := range seedList.Items {
```

Comment:
what do you think about the following algorithm. It will work faster:

```
shootsPerSeed := make(map[string]int)
	for _, shoot := range shootList.Items {
		shootsPerSeed[*shoot.Spec.Cloud.Seed]++
	}
	for seed, numberOfShoots := range shootsPerSeed {
		fmt.Printf("    %s: %d \n", seed, numberOfShoots)
	}
```

Reply:
Nice suggestion. Applied it will little modification for sorting - to have always the seed seeds in output sorted (in the same order).